src/main/java/nl/tudelft/jpacman/level/PlayerCollisionsSuperPellet.java

```diff
@@ -0,0 +1,65 @@
+package nl.tudelft.jpacman.level;
+
+import nl.tudelft.jpacman.board.Unit;
+import nl.tudelft.jpacman.npc.ghost.Ghost;
+import nl.tudelft.jpacman.npc.ghost.VulnerableGhost;
+
+/**
+ * @author Corentin Ducruet
+ */
+public class PlayerCollisionsSuperPellet extends PlayerCollisions {
+
+    private LevelSuperPellet level;
+
+    public  void addLevel(LevelSuperPellet l){
+        level=l;
+    }
+    @Override
+    public void collide(Unit mover, Unit collidedOn) {
+        if (mover instanceof Player) {
+            if (collidedOn instanceof Ghost) {
+                playerVersusGhost((Player) mover, (VulnerableGhost) collidedOn);
+            }
+
+            if (collidedOn instanceof SuperPellet) {
+                playerVersusSuperPellet((Player) mover, (SuperPellet) collidedOn);
+            }
+            else if (collidedOn instanceof Pellet) {
+                playerVersusPellet((Player) mover, (Pellet) collidedOn);
+            }
+        }
+        else if (mover instanceof Ghost) {
+            if (collidedOn instanceof Player) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2533232592/issues/source?bid=3256381&fileBranchId=3284358#l32)